### PR TITLE
feat: do not interfere with shared workers

### LINKED ISSUE #1350: can't handle request for shared workers
### Prerequisites

- [X] I confirm my issue is not in the [opened issues](https://github.com/mswjs/msw/issues)
- [X] I confirm the [Frequently Asked Questions](https://mswjs.io/docs/faq) didn't contain the answer to my issue

### Environment check

- [X] I'm using the [latest](https://github.com/mswjs/msw/releases/latest) `msw` version
- [X] I'm using Node.js version 14 or higher

### Browsers

Chromium (Chrome, Brave, etc.)

### Reproduction repository

not available

### Reproduction steps

When trying to request a shared worker, the clientId attached to fetch event object is an empty string.

That causes the following line to break under mockServiceWorker.js in resolveMainClient function:
`if(client.frameType === 'top-level')`

I managed to make it work currently by checking for client like the following:
`if(client && client.frameType === 'top-level')`

With that I am able to work with shared worker.

Do you have any ideas why that happens?

msw version: 0.44.2

Thanks

### Current behavior

the msw fails handling the request

### Expected behavior

msw should handle the request.

### LINKED ISSUE #1354: Can't handle requests from a shared worker
### Prerequisites

- [X] I confirm my issue is not in the [opened issues](https://github.com/mswjs/msw/issues)
- [X] I confirm the [Frequently Asked Questions](https://mswjs.io/docs/faq) didn't contain the answer to my issue

### Environment check

- [X] I'm using the [latest](https://github.com/mswjs/msw/releases/latest) `msw` version
- [X] I'm using Node.js version 14 or higher

### Browsers

Chromium (Chrome, Brave, etc.)

### Reproduction repository

https://github.com/njoel24/integration-test-ms-shared-worker

### Reproduction steps

Hi @kettanaito, I  created a new issue with requested info from you.
I don't know if you get notifications on closed issue, and I just created a new one, just to be sure.

Here the issue:

When trying to request a shared worker, the clientId attached to fetch event object is an empty string.

That causes the following line to break under mockServiceWorker.js in resolveMainClient function:
if(client.frameType === 'top-level')

I managed to make it work currently by checking for client like the following:
if(client && client.frameType === 'top-level')

With that I am able to work with shared workers.

In the issue I described there are no iframes involved.

Here an integration test to reproduce the issue: https://github.com/njoel24/integration-test-ms-shared-worker
Just clone, install, run 'npm run build' and run 'npm run start' 

Here a screenshot of the error:

test results

Please note that the handler has not been implemented and left as null because the error happens before.

Grateful if you can check it.
I would be more than happy to give you more information and contribute as well.

msw version: 0.44.2

Thanks

### Current behavior

the msw fails handling the request

### Expected behavior

msw should handle the request.

### PR DESCRIPTION
**Pull request by mucsi96**
- Closes #1350
- Fixes #1354 

Here I am adding support for shared workers.

The issue is that in case of shared workers the request event doesn't contain `clientId`.
Due to that the `client` is `undefined` which breaks the code here

```tsx
if (client.frameType === 'top-level') {
    return client
  }
```

### CODE REVIEW

**Review by kettanaito — DISMISSED** [review]
This looks fantastic! Thank you for adding this support, @mucsi96 🎉 

Let me try this out and get back to you.

File: test/msw-api/setup-worker/scenarios/shared-worker/shared-worker.test.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import { waitFor } from '../../../../support/waitFor'
+
+function createRuntime() {
+  return pageWith({
+    example: path.resolve(__dirname, 'shared-worker.mocks.ts'),
+    contentBase: path.resolve(__dirname),
+  })
+}
+
+test('supports shared workers', async () => {
```
Comment (kettanaito) [inline test/msw-api/setup-worker/scenarios/shared-worker/shared-worker.test.ts]: Do I understand it correctly that the intention is not to intercept requests from shared workers in any way but rather not obstruct them if they do exist alongside MSW?

If it's the latter, we should reflect that in the test name.

Comment (mucsi96) [reply test/msw-api/setup-worker/scenarios/shared-worker/shared-worker.test.ts]: Sorry I am not sure I get your question. Please adjust the test name as you find it better. 

This test just tests that shared workers are not broken alongside MSW. Without the change in service worker this test fails.

Comment (mucsi96) [reply test/msw-api/setup-worker/scenarios/shared-worker/shared-worker.test.ts]: Intercepting the worker code itself also works with MSW after that. But that's working out of the box if this is fixed.

Comment (kettanaito) [reply test/msw-api/setup-worker/scenarios/shared-worker/shared-worker.test.ts]: Thanks for clarifying!

**Review by kettanaito — APPROVED** [review]

**Comment by kettanaito** [comment]
I'd like to group this with #1445 since both changes are potentially breaking and would fall nicely into a single minor release.

**Comment by kettanaito** [comment]
## Released: v0.48.0 🎉

This has been released in v0.48.0!

- 📄 [**Release notes**](https://github.com/mswjs/msw/releases/tag/v0.48.0)
- 📦 [npm package](https://www.npmjs.com/package/msw/v/0.48.0)

Make sure to always update to the latest version (`npm i msw@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.

**Pull request merged**